google-datacatalog-rdbms-connector/src/google/datacatalog_connectors/rdbms/prepare/datacatalog_tag_template_factory.py

```diff
@@ -98,24 +98,52 @@ def make_tag_template_for_table_metadata(self):
         table_container_type = self.__metadata_definition[
             'table_container_def']['type']
 
-        table_container_name_field = datacatalog.TagTemplateField()
-        table_container_name_field.type.primitive_type = self.__STRING_TYPE
-        table_container_name_field.display_name = '{} Name'.format(
-            table_container_type.capitalize())
-
-        tag_template.fields[table_container_name] = table_container_name_field
+        self._add_primitive_type_field(
+            tag_template, table_container_name, self.__STRING_TYPE,
+            '{} Name'.format(table_container_type.capitalize()))
 
         self.__add_database_name_to_tag_template(tag_template)
 
-        num_rows_field = datacatalog.TagTemplateField()
-        num_rows_field.type.primitive_type = self.__DOUBLE_TYPE
-        num_rows_field.display_name = 'Number of rows'
-        tag_template.fields['num_rows'] = num_rows_field
+        self._add_primitive_type_field(tag_template, 'num_rows',
+                                       self.__DOUBLE_TYPE, 'Number of rows')
+
+        self._add_primitive_type_field(tag_template, 'table_size_MB',
+                                       self.__DOUBLE_TYPE, 'Table Size in MB')
+
+        self._add_primitive_type_field(tag_template, 'table_type',
+                                       self.__STRING_TYPE, 'Table type')
+
+        self._add_primitive_type_field(tag_template, 'has_primary_key',
+                                       self.__BOOL_TYPE, 'Has primary key')
+
+        return tag_template_id, tag_template
+
+    def make_tag_template_for_column_metadata(self):
+        """Create a Tag Template with technical fields for Column metadata.
+         :return: tag_template_id, tag_template
+        """
+
+        tag_template = datacatalog.TagTemplate()
+
+        column_type = self.__metadata_definition['column_def']['type']
+
+        tag_template_id = '{}_{}_metadata'.format(self.__entry_group_id,
+                                                  column_type)
+
+        tag_template.name = \
+            datacatalog.DataCatalogClient.tag_template_path(
+                project=self.__project_id,
+                location=self.__location_id,
+                tag_template=tag_template_id)
+
+        tag_template.display_name = '{} {} - Metadata'.format(
+            self.__entry_group_id.capitalize(), column_type.capitalize())
+
+        self._add_primitive_type_field(tag_template, 'mask', self.__BOOL_TYPE,
```

Comment:
Aren't the mask and mask_expression fields specific to SAP Hana columns? If so, maybe it is better to handle them in the SAP Hana connector. WDYT? 🤔

Reply:
Same answer from tag factory.